shells/pipes-shell-2/api/autofill.js

```diff
@@ -26,7 +26,8 @@ export const autofill = async (msg, tid, bus, composerFactory, storage, context)
     } else {
       // arc
       const composer = composerFactory(msg.modality);
-      const arc = await Utils.spawn({id: generateId(), composer, context/*, storage*/});
+      const id = generateId();
+      const arc = await Utils.spawn({id, composer, context/*, storage*/, ports: [portFactory(id, bus)]});
```

Comment:
Do Arcs require `ports` array at construction time (in theory)? It would be easier if one could construct ports on an Arc, because the Arc knows it's own id.